File: 2024/AdventOfCode/Tests/Day07Test.cs
using NUnit.Framework;

namespace AdventOfCode.Tests;

[TestFixture]
public class Day07Test {
    private const string Example = "Inputs/07-Example.txt";
    private const string Input = "Inputs/07.txt";
    private static Day07 _day;

    [TestFixture]
    public class Part1 {
        [Test]
        public void TestExample() {
            _day = new Day07(Example);
            const string expected = "3749";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day07(Input);
            const string expected = "1260333054159";
            var solution = _day.Solve_1();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }

    [TestFixture]
    public class Part2 {
        [Test]
        public void TestExample() {
            _day = new Day07(Example);
            const string expected = "11387";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }

        [Test]
        public void TestInput() {
            _day = new Day07(Input);
            const string expected = "162042343638683";
            var solution = _day.Solve_2();
            Assert.That(solution.Result, Is.EqualTo(expected));
        }
    }
}

File: 2025/test/advent_of_code/day_07_test.exs
defmodule AdventOfCode.Day07Test do
  use ExUnit.Case

  import AdventOfCode.Day07

  @day 7
  @year 2025

  @example String.trim("""
           .......S.......
           ...............
           .......^.......
           ...............
           ......^.^......
           ...............
           .....^.^.^.....
           ...............
           ....^.^...^....
           ...............
           ...^.^...^.^...
           ...............
           ..^...^.....^..
           ...............
           .^.^.^.^.^...^.
           ...............
           """)

  test "part1 example" do
    result = part1(@example)

    assert result == 21
  end

  test "part1" do
    input = AdventOfCode.Input.get!(@day, @year)
    result = part1(input)

    assert result == 1490
  end

  test "part2 example" do
    result = part2(@example)

    assert result == 40
  end

  test "part2" do
    input = AdventOfCode.Input.get!(@day, @year)
    result = part2(input)

    assert result == 3_806_264_447_357
  end
end
